### PR TITLE
Ports TG's Jukebox

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports TG's jukebox, with some modifications.

TG's jukebox does the same thing as the disco machine, but it doesn't have sparkly lights or make you dance.

There's a subtype of jukebox that does the same, exact thing, but also has the sparkly lights everyone knows and loves.

I did not implement TG's system where you can put sounds in a specific music folder to stream to players, for the simple reason that it caused major lag and performance issues. As such, until someone can figure out how to integrate this into goonchat's HMTL5 audio streaming capabilities, it's going to remain fairly static (sadly).

Changes to discoballs/jukeboxes:
- Removes the ability to generate random sound effects from the disco ball
- Jukebox/disco ball sound will now obey musical instrument sound preferences
- I went ahead and added a bunch of existing songs to the jukebox/disco ball. These were already compiled into the code, so they might as well be there

Differences from TG: I didn't port their crappy jukebox sprites, because those things were horrendous.

Maybe we could map this into the bar? Who knows

:cl: Fox McCloud
tweak: Splits disco balls into jukeboxes and discoballs; the former only plays music.
tweak: Disco balls no longer have random sound effects
/:cl:

### CODE REVIEW

File: code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -476,5 +466,17 @@
...
 		playsound(src,'sound/machines/terminal_off.ogg',50,1)
-		icon_state = "disco0"
+		update_icon()
 		stop = world.time + 100
+
+
+/obj/machinery/jukebox/disco/process()
+	..()
+	if(active)
+		for(var/mob/M in rangers)
```
Comment (Citinited) [inline code/game/machinery/dance_machine.dm]: Does this make ghosts and other weird mobs dance?

**Comment by taukausanake** [comment]
What a glorious day. The jukebox has returned! Shame you don't get music from it just yet but still good

File: code/game/machinery/dance_machine.dm (hunk 1 above)
Comment (Fox-McCloud) [reply code/game/machinery/dance_machine.dm]: yup!

Comment (Citinited) [reply code/game/machinery/dance_machine.dm]: I might be being dumb but don't you need to tab `dance(M)` an additional time to have it be included in `isliving(M)` and therefore be checking for nonliving mobs?

**Comment by variableundefined** [comment]
@Fox-McCloud Oofs, conflicts on this.

**Comment by Kyep** [comment]
If I'm reading this right, this makes BOTH disco ball and jukebox no longer play their sounds at all for clients with instrument preferences off? If so, that's great.

File: code/game/machinery/dance_machine.dm [hunk 2]
```diff
@@ -70,25 +78,32 @@
...
+/obj/machinery/jukebox/interact(mob/user)
+	if(!Adjacent(user) && !isAI(user))
+		return
 	if(!anchored)
 		to_chat(user,"<span class='warning'>This device must be anchored by a wrench!</span>")
 		return
-	if(!Adjacent(user) && !isAI(user))
+	if(!allowed(user))
+		to_chat(user,"<span class='warning'>Error: Access Denied.</span>")
+		user.playsound_local(src,'sound/misc/compiler-failure.ogg', 25, 1)
```
Comment (farie82) [inline code/game/machinery/dance_machine.dm:102]: spacing here and below

File: code/game/machinery/dance_machine.dm [hunk 3]
```diff
@@ -436,35 +429,32 @@
...
-
-/obj/machinery/disco/process()
-	if(charge < 35)
-		charge += 1
+/obj/machinery/jukebox/process()
 	if(world.time < stop && active)
 		var/sound/song_played = sound(selection.song_path)
 
 		for(var/mob/M in range(10,src))
+			if(!M.client || !(M.client.prefs.sound & SOUND_INSTRUMENTS))
```
Comment (farie82) [inline code/game/machinery/dance_machine.dm:453]: wasn't there a preference already to stop this?

File: code/game/machinery/dance_machine.dm (hunk 1 above)
Comment (farie82) [reply code/game/machinery/dance_machine.dm:476]: What Cit said. Currently you pass non livings as well while `dance()` expects `mob/living`

**Comment by taukausanake** [comment]
Coming up on the one year anniversary here

File: code/game/machinery/dance_machine.dm (hunk 1 above)
Comment (Crazylemon64) [reply code/game/machinery/dance_machine.dm:476]: I concur with the above. Please indent the dance call to avoid calling it on living mobs

**Comment by Fox-McCloud** [comment]
going to close this; there's some more important things, at the moment--might re-open this one day--if someone else picks it up, I won't care.

**Pull request closed without merging**